eos-update-flatpak-repos

```diff
@@ -1332,7 +1343,22 @@ def copy_commit(repo, src_rev, dest_ref,
                                                    commit_time)
     logging.info('Created new commit %s', dest_checksum)
 
-    return dest_checksum, vendor_prefixes
+    return dest_checksum, vendor_prefixes, migrate_extra
+
+
+def migrate_extra_data(old_ref, new_ref):
+    old_deploy = old_ref.get_deploy_dir()
+    old_extra = os.path.join(old_deploy, 'files', 'extra')
+
+    new_deploy = new_ref.get_deploy_dir()
+    new_extra = os.path.join(new_deploy, 'files', 'extra')
+
+    if os.path.exists(new_extra):
+        raise Exception('not expecting {} to already have extra data dir'
+                        .format(new_ref))
+
+    logging.debug('Copying extra data from {} to {}'.format(old_ref, new_ref))
+    subprocess.check_call(['cp', '-al', old_extra, new_extra])
```

Comment:
Of course, since this just creates a small handful of directories and hardlinks, it's very fast so the window of error is very small.